### ISSUE TITLE
Permissions check for registered user should pass if anonymous user has permission

### ISSUE THREAD

**Issue by JDougherty** [django-guardian/django-guardian#391]
I'm not sure if this is intentional or not, or if there's a documented or recommended way to do what I'd like to see, but let me try to explain. I feel that if the anonymous user has some permission for an object, then a permission check for a registered/authorized user should succeed even if the registered used doesn't explicitly have that permission, but that is currently not the case. In my usage, I can't imagine a scenario where I would want an anonymous user to be able to do something that a registered user should not be able to do. 

I suppose there may be a reason why the current behavior is the current behavior, so I'm not necessarily requesting that the default behavior be changed, but it would be nice to have a built-in way of doing what I describe above, or at least documenting it somewhere. For now, I've sub-classed PermissionRequiredMixin to do what I want:

```python
class PermissionRequiredMixinWithAnonymous(PermissionRequiredMixin):
    def check_permissions(self, request):
        forbidden = super(PermissionRequiredMixinWithAnonymous, self).check_permissions(request)
        if forbidden:
            perms = self.get_required_permissions(request)
            anon = get_anonymous_user()
            obj = self.get_permission_object()
            has_permissions = all(anon.has_perm(perm, obj) for perm in perms)
            if has_permissions:
                forbidden = None
        return forbidden
```

I wanted to make this issue to get some insight from the developers/maintainers of this project and see if this functionality (not necessarily the above implementation) would be useful to include in guardian, or maybe even if there's some other recommended way of doing what I want to do. If there isn't a way to do it now, and you do feel it would be useful, I would be interested in implementing the functionality and submitting a pull request.

**Comment by Pesticles** [comment]
I think they way you've done it is the right way, subclassing the Mixin.

I think the current default behaviour is correct, I believe it would violate the principle of least surprise if your way was the default, however I can certainly see that your way would be valid/useful for many people.

**Comment by JDougherty** [comment]
Yeah certainly, I mentioned in the first post that I don't necessarily think the default behavior should be changed. Rather, since I think this is probably a common use case, it might be nice to have a call out in the documentation and possibly have some example code or even implement this functionality within django-guardian itself.
